python/lsst/meas/base/measurementInvestigationLib.py

```diff
@@ -0,0 +1,125 @@
+#
+# LSST Data Management System
+#
+# Copyright 2008-2017  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+from collections import Iterable
+
+from lsst.afw.table import SourceCatalog
+from lsst.meas.base import NoiseReplacer, NoiseReplacerConfig
+
+
+def rebuildNoiseReplacer(exposure, measCat):
+    """ Recreate NoiseReplacer used in measurement
+
+    Given a measurement catalog and the exposure on which the measurements were
+    made, reconstruct the NoiseReplacer object that was used to mask out
+    sources durring measuremnt.
+
+    Parameters
+    ----------
+    exposure : lsst.awf.exposure.Exposure
+        The exposure on which measurements were made
+
+    meaCat : lsst.afw.table.SourceCatalog
+        Catalog containing the outputs of measurements on each source
+
+    Returns
+    -------
+    noiseReplacer : lsst.meas.base.NoiseReplacer
+        Object used to replace and or restore sources in the exposure with
+        deterministic noise
+    """
+
+    algMetadata = measCat.getMetadata()
+    noiseReplacerConf = NoiseReplacerConfig()
+    noiseReplacerConf.noiseSeedMultiplier = \
+        algMetadata.get("NOISE_SEED_MULTIPLIER")
```

Comment:
Made these class level variables in single frame measurement task, and referenced them here